packages/shared/src/react/hooks/useReverification.ts

```diff
@@ -68,26 +98,55 @@ function createReverificationHandler(params: { onOpenModal: Clerk['__experimenta
       }
 
       return result;
-    }) as Fetcher;
+    }) as ExcludeClerkError<Awaited<ReturnType<Fetcher>>, Parameters<Fetcher>[1]>;
   }
 
   return assertReverification;
 }
 
-function __experimental_useReverification<Fetcher extends () => Promise<any>>(fetcher: Fetcher): readonly [Fetcher] {
+type UseReverificationResult<
+  Fetcher extends (...args: any[]) => Promise<any> | undefined,
+  Options extends UseReverificationOptions,
+> = readonly [(...args: Parameters<Fetcher>) => Promise<ExcludeClerkError<Awaited<ReturnType<Fetcher>>, Options>>];
+
+/**
+ * Receives a fetcher async function and returned an enhanced fetcher that automatically handles the reverification flow
+ * by displaying a prebuilt UI component when the request from the fetcher fails with a reverification error response.
+ *
+ * While the UI component is displayed the promise is still pending.
+ * On success: the original request is retried one more time.
+ * On error:
+ * (1) by default the fetcher will return `null` and the `onCancel` callback will be executed.
+ * (2) when `throwOnCancel: true` instead of returning null, the returned fetcher will throw a `ClerkRuntimeError`.
+ *
+ * @example
+ * A simple example:
+ *
+ * function Hello() {
+ *   const [fetchBalance] = useReverification(()=> fetch('/transfer-balance',{method:"POST"}));
+ *   return <button onClick={fetchBalance}>...</button>
+ * }
+ */
+function __experimental_useReverification<
+  Fetcher extends (...args: any[]) => Promise<any> | undefined,
+  Options extends UseReverificationOptions,
+>(fetcher: Fetcher, options?: Options): UseReverificationResult<Fetcher, Options> {
   const { __experimental_openUserVerification } = useClerk();
   const fetcherRef = useRef(fetcher);
+  const optionsRef = useRef(options);
 
   const handleReverification = useMemo(() => {
     const handler = createReverificationHandler({
-      onOpenModal: __experimental_openUserVerification,
+      openUIComponent: __experimental_openUserVerification,
+      ...optionsRef.current,
     })(fetcherRef.current);
     return [handler] as const;
-  }, [__experimental_openUserVerification, fetcherRef.current]);
+  }, [__experimental_openUserVerification, fetcherRef.current, optionsRef.current]);
```

Comment:
Adding the references in here, will not cause the `useMemo` to re-render explicitly, but if something else causes the Component that the hook is mounted on, this will use the new references if they changed